src/cloudai/schema/test_template/nemo_launcher/slurm_command_gen_strategy.py

```diff
@@ -58,6 +58,10 @@ def gen_exec_command(
         for key, value in final_env_vars.items():
             self.final_cmd_args[f"env_vars.{key}"] = value
         self.final_cmd_args["cluster.partition"] = self.slurm_system.default_partition
+        reservation_key = "--reservation "
+        if self.slurm_system.extra_srun_args and reservation_key in self.slurm_system.extra_srun_args:
+            reservation = extra_cmd_args.split(reservation_key, 1)[1].split(" ", 1)[0]
```

Comment:
Should it be `self.slurm_system.extra_srun_args` not `extra_cmd_args`?

Anyway, it seems that we can add `extra_srun_args = "--reservation name"` into a System TOML and also add `extra_cmd_args = "+cluster.reservation=name"` into a Test TOML? This way we don't need to change python code.

Reply:
yes thanks for noticing the error

well the problem with adding it in the test scenario is that it's not part of the test, it's dependant on a specific run.

What do you think @lappazos @srinivas212 ?

Reply:
`extra_cmd_args` is part of Test TOML, not Scenario TOML. So that means you'll have to modify multiple files if follow my suggestion, which is not user-friendly.

Let's go with your solution, but please expand `TestNeMoLauncherSlurmCommandGenStrategy__GenExecCommand` tests to cover your case.

Reply:
right I meant to say test not test scenario, my bad.

Alright sure I'll try to cover it today

Reply:
> `extra_cmd_args` is part of Test TOML, not Scenario TOML. So that means you'll have to modify multiple files if follow my suggestion, which is not user-friendly.
> 
> Let's go with your solution, but please expand `TestNeMoLauncherSlurmCommandGenStrategy__GenExecCommand` tests to cover your case.

updated